#!/usr/bin/env python
PACKAGE = "proc_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("CONSTANT_DEPTH_FORCE",double_t, 0, "Constant effort to keep the submarine depth", 0, -100.0, 100.0)

gen.add("Z_P", double_t, 0, "Z: The proportional coefficient", 0, -100.0, 100.0)
gen.add("Z_I", double_t, 0, "Z: The integration coefficient", 0, -100.0, 100.0)
gen.add("Z_D", double_t, 0, "Z: The derivative coefficient", 0, -100.0, 100.0)
gen.add("Z_I_LIMIT", double_t, 0, "Z: Maximum integration", 0, -100.0, 100.0)
gen.add("Z_MIN_ACTUATION", double_t, 0, "Z: Minimum actuation", 0, -100.0, 100.0)
gen.add("Z_MAX_ACTUATION", double_t, 0, "Z: Maximum actuation", 0, -100.0, 100.0)

exit(gen.generate(PACKAGE, "proc_control", "PIDZAxis"))
